app/assets/javascripts/Components/markdown_edit.jsx

```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import MarkdownPreview from "./markdown_preview";
+import {Tab, Tabs, TabList, TabPanel} from "react-tabs";
+
+import PropTypes from "prop-types";
+
+export default class MarkdownEdit extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      content: this.props.content,
+      annotation_text_id: "",
+    };
+  }
+
+  handleChange = event => {
+    this.setState({content: event.target.value});
+    this.props.handleChange(event);
+  };
+
+  render() {
+    return (
+      <Tabs disableUpDownKeys>
+        <TabList>
+          <Tab id="write-tab">{I18n.t("write")}</Tab>
+          <Tab id="preview-tab">{I18n.t("preview")}</Tab>
+        </TabList>
+        <TabPanel forceRender>
+          <label>
+            <textarea
+              required={true}
+              id="new_annotation_content"
+              name="content"
+              placeholder={I18n.t("results.annotation.placeholder")}
+              value={this.state.content}
+              onChange={this.handleChange}
+              rows="8"
+              autoFocus={true}
+            />
+          </label>
+
+          <div className={this.props.show_autocomplete ? "" : "hidden"}>
+            <ul className="tags" key="annotation_completion" id="annotation_completion">
```

Comment:
Similarly, it's good to have this autocompletion subcomponent here since it might be nice to use it more in the future, but its classes should be renamed to avoid specific mention of annotations. Also remove the `id`s here, since `id`s need to be unique but in principle we could have more than one instance of this component rendered on the page at a time.

Reply:
I removed all unnecessary ids and made the necessary ones props :+1: